docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,290 @@
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one or more
+ contributor license agreements.  See the NOTICE file distributed with
+ this work for additional information regarding copyright ownership.
+ The ASF licenses this file to You under the Apache License, Version 2.0
+ (the "License"); you may not use this file except in compliance with
+ the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+<script><!--#include virtual="../../js/templateData.js" --></script>
+
+<script id="content-template" type="text/x-handlebars-template">
+    <!-- h1>Developer Guide for Kafka Streams</h1 -->
+    <div class="sub-nav-sticky">
+        <div class="sticky-top">
+            <!-- div style="height:35px">
+              <a href="/{{version}}/documentation/streams/">Introduction</a>
+              <a class="active-menu-item" href="/{{version}}/documentation/streams/developer-guide">Developer Guide</a>
+              <a href="/{{version}}/documentation/streams/core-concepts">Concepts</a>
+              <a href="/{{version}}/documentation/streams/quickstart">Run Demo App</a>
+              <a href="/{{version}}/documentation/streams/tutorial">Tutorial: Write App</a>
+            </div -->
+        </div>
+    </div>
+
+    <div class="section" id="naming">
+        <span id="streams-developer-guide-dsl-topology-naming"></span>
+        <h1>Naming Kafka Streams DSL Appliations<a class="headerlink" href="#naming" title="Permalink to this headline"></a></h1>
+
+
+		With the 2.4 release, you now can give names to processors when using the Kafka 
+		Streams DSL.  Unlike the PAPI, where you are required to give a name for the processor, in the 
+		DSL, the names are generated for you.  The name generation strategy is 
+		<code>KSTREAM|KTABLE-&gt;operator name&lt;-&gt;number suffix&lt;</code>. The number suffix is a 
+		globally incrementing number that represents the operator's order in the topology.  
+		The number prefixed with a varying number of "0"s to create a suffix that is consistently 10 
+		characters long.   While the internal naming makes creating a topology with the DSL much more straightforward, 
+		there are a couple of trade-offs.  The first trade-off is what we could consider cognitive, and the other 
+		more severe trade-off is the shifting of repartition topic, 
+		changelog topic, and state store names.
+
+
+		<h2>Cognitive Issues</h2>
```

Comment:
Have you considered any alternative names for this section/trade-off? "Cognitive Issues" seems to have a weird connotation, what about just `Readability` or `Readability Problems`?